### PR TITLE
ER IDETECT-3046 - Add support for [project-name].nuget.g.props to

### PR DESCRIPTION
**Pull request by gopannabd**
Add support for [project-name].nuget.g.props to extract referenced Project asset JSON file path.
Directory.Build.props is not included as it falls under centralized configuration management (for solutions) feature of MSBuild.

Resolve IDETECT-3046

### CODE REVIEW

File: detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/SolutionDirectoryBuildPropertyLoader.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+        public SolutionDirectoryBuildPropertyLoader(string propertyPath, NugetSearchService nugetSearchService)
+        {
+            PropertyPath = propertyPath;
+            NugetSearchService = nugetSearchService;
+        }
+
+        public HashSet<PackageId> Process()
+        {
+            var tree = new NugetTreeResolver(NugetSearchService);
```
Comment (andrian-synopsys) [inline detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/SolutionDirectoryBuildPropertyLoader.cs:25]: Is this line necessary?

File: detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/ProjectNugetgPropertyLoader.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        public ProjectNugetgPropertyLoader(string propertyPath, NugetSearchService nugetSearchService)
+        {
+            PropertyPath = propertyPath;
+            NugetSearchService = nugetSearchService;
+        }
+
+        public string Process()
+        {
+            var tree = new NugetTreeResolver(NugetSearchService);
```
Comment (andrian-synopsys) [inline detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/ProjectNugetgPropertyLoader.cs:24]: Is this line necessary?

File: detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/SolutionDirectoryBuildPropertyLoader.cs (hunk 1 above)
Comment (gopannabd) [reply detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/SolutionDirectoryBuildPropertyLoader.cs:25]: Will remove, thanks!

File: detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/ProjectNugetgPropertyLoader.cs (hunk 2 above)
Comment (gopannabd) [reply detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/ProjectNugetgPropertyLoader.cs:24]: Will remove, thanks!

**Review by shantyk — APPROVED** [review]

**Review by shantyk — APPROVED** [review]

**Review by andrian-synopsys — APPROVED** [review]

**Pull request merged**